docs/content/reference/konnectivity.md

```diff
@@ -0,0 +1,393 @@
+curl -x "https://127.0.0.1:8090" \
+  --proxy-cacert ${workdir}/konnectivity_ca.crt \
+  --proxy-cert ${workdir}/client.crt \
+  --proxy-key ${workdir}/client.key \
+  --cacert ${workdir}/kubelet_ca.crt \
+  --cert ${workdir}/kubelet_client.crt \
+  --key ${workdir}/kubelet_client.key \
+  "https://\${nodeip}:10250/metrics"
+
+# Kill the port-forward job
+kill %1
+```
+
+## Konnectivity Q&A
+
+**Upstream allows grpc or http-connect mode for the konnectivity protocol, why does HyperShift use http-connect?**
+We need to use http-connect to allow clients other than the Kubernetes
+API server to send traffic through the konnectivity server. The socks5
+and http proxies can use http-connect to establish a connection through
+konnectivity-server.
+
+**Upstream allows connecting to the konnectivity server via unix domain socket or via tcp port. Why does HyperShift use tcp port?**
+The konnectivity server only allows one way of serving (via UDS or TCP
+port). We need to use a TCP port so we can expose it as a service to
+clients that use the socks5 and http proxies.
+
+**Why does Konnectivity server run as a sidecar of the Kubernetes API server and not in its own deployment?**
+When running as a sidecar, the Kubernetes API server can communicate
+with the Konnectivity server very fast, since traffic only goes through
+the same pod's interface. It also allows the Kubernetes API server to
+communicate with a konnectivity-server as soon as it comes up.
+
+**How is the Konnectivity server secured in HyperShift?**
+We use mTLS to authenticate connections to Konnectivity either from
+clients like Kubernetes API server or from agents. The certificates used
+for authentication are in the konnectivity-client and konnectivity-agent
+secrets in the control plane.
+
+**How does Konnectivity HA work?**
+When a HyperShift control plane runs in HA mode, there are by default 3
+replicas of the konnectivity server (1 sidecar per kube-apiserver pod).
+Each server generates an ID when starting up and the number of total
+servers is passed as a command line argument. When an agent dials the
+konnectivity server endpoint, the server's response includes its own ID
+and the total number of servers that the agent can expect. If greater
+than one, the agent keeps dialing the konnectivity server endpoint until
+it can establish connections with the same number of unique servers
+(identified by their unique id).
+
+**What breaks if Konnectivity is not working?**
+The first thing that breaks when Konnectivity is not working is any
+operation that requires that the Kubernetes API server access kubelets
+on nodes. This includes getting logs from pods, proxying to
+pods/services, port-forwarding and remote execution. The next thing that
+breaks is any aggregated API servers and webhooks that use data plane
+services.
+
+Things that are not impacted by Konnectivity being down is the ability
+to run pods or the ability of nodes to communicate with the control
+plane or each other.
```

Comment:
I'm not sure if this last paragraph adds value becuase _most_  things work, even when konnectivity is down (unless you have create webhooks on basic resources or something)

Reply:
removed